source/common/http/conn_manager_impl.cc

```diff
@@ -1477,16 +1477,16 @@ bool ConnectionManagerImpl::ActiveStream::verbose() const {
 
 void ConnectionManagerImpl::ActiveStream::callHighWatermarkCallbacks() {
   ++high_watermark_count_;
-  if (watermark_callbacks_) {
-    watermark_callbacks_->onAboveWriteBufferHighWatermark();
+  for (auto watermark_callbacks : watermark_callbacks_) {
```

Comment:
Can you add explicit HCM unit tests for these changes? cc @alyssawilk to also look through this part.

Reply:
Also E2E tests if you can.  Some of the large body integration test with low buffer limits are worth checking out.

Reply:
Any particular cases you'd like to see covered? I'm not very familiar with the buffer watermark lifecycle.